pkg/setup/setup_test.go

```diff
@@ -22,12 +22,14 @@ func TestValidateCanRunTargetArchTrivial(t *testing.T) {
 }
 
 func TestValidateCanRunTargetArchUnsupportedCanary(t *testing.T) {
-	var logbuf bytes.Buffer
-	logrus.SetOutput(&logbuf)
+	logbuf := new(bytes.Buffer)
```

Comment:
I'm curious (and its not important), any particular reason to go from `var logbuf bytes.Buffer` to `logbuf := new(bytes.Buffer)` ? Is there a technical (or other) advantage or just (personal) preference? No need to change it, I don't mind - just curious if there is something to learn here :) 

I mean, from a purely technical PoV having `var logbuf` on the stack seems "cheaper" because no malloc() call is require and the GC does not need to track it (as its just on the stack and auto cleaned).

Reply:
Good catch, no, no reason. Escape analysis is a strong feature we should always take advantage of. Removed here and there (total 2 places).